meta/AggregationCloud.php

```diff
@@ -0,0 +1,206 @@
+     *
+     * @param string $id
+     * @param string $mode
+     * @param \Doku_Renderer $renderer
+     * @param SearchConfig $searchConfig
+     */
+    public function __construct($id, $mode, \Doku_Renderer $renderer, SearchCloud $searchConfig) {
+        $this->id = $id;
+        $this->mode = $mode;
+        $this->renderer = $renderer;
+        $this->searchConfig = $searchConfig;
+        $this->data = $searchConfig->getConf();
+        $this->columns = $searchConfig->getColumns();
+        $this->result = $this->searchConfig->execute();
+        $this->resultCount = $this->searchConfig->getCount();
+
+        $this->max = $this->result[0]['count'];
+        $this->min = end($this->result)['count'];
+    }
+
+    /**
+     * Create the cloud on the renderer
+     */
+    public function render() {
+
+        $this->sortResults();
+
+        $this->startScope();
+        $this->startList();
+        foreach ($this->result as $result) {
+            $this->renderTag($result);
+        }
+        $this->finishList();
+        $this->finishScope();
+        return;
+    }
+
+    /**
+     * Adds additional info to document and renderer in XHTML mode
+     *
+     * @see finishScope()
+     */
+    protected function startScope() {
+        // wrapping div
+        if($this->mode != 'xhtml') return;
+        $this->renderer->doc .= "<div class=\"structcloud\">";
+    }
+
+    /**
+     * Closes the table and anything opened in startScope()
+     *
+     * @see startScope()
+     */
+    protected function finishScope() {
+        // wrapping div
+        if($this->mode != 'xhtml') return;
+        $this->renderer->doc .= '</div>';
+    }
+
+    /**
+     * Render a tag of the cloud
+     *
+     * @param ['tag' => Value, 'count' => int] $result
+     */
+    protected function renderTag($result) {
+        /**
+         * @var Value $value
+         */
+        $value = $result['tag'];
+        $count = $result['count'];
+        if ($value->isEmpty()) {
+            return;
+        }
+
+        $this->renderer->listitem_open(1);
+        $this->renderer->listcontent_open();
+
+        $this->renderTagLink($value, $count);
+        $this->renderer->listcontent_close();
+        $this->renderer->listitem_close();
+    }
+
+    /**
+     * @param Value $value
+     * @param int $count
+     */
+    protected function renderTagLink(Value $value, $count) {
+        $type = strtolower($value->getColumn()->getType()->getClass());
+        $weight = $this->getWeight($count, $this->min, $this->max);
+        $schema = $this->data['schemas'][0][0];
+        $col = $value->getColumn()->getLabel();
+
+        if (!empty($this->data['target'])) {
+            $target = $this->data['target'];
+        } else {
+            global $INFO;
+            $target = $INFO['id'];
+        }
+
+        $tagValue = $value->getDisplayValue();
+        if (empty($tagValue)) {
+            $tagValue = $value->getRawValue();
```

Comment:
There is an edge-case for `page` types with `usetitles` set and pages that never existed or uncleaned data after the column has been converted to `page` type. But maybe that is a bug and should be fixed?